wrappers/python/tests/TestPdbReporter.py

```diff
@@ -0,0 +1,307 @@
+            atomSubset = list(range(pdb.topology.getNumAtoms()))
+            simulation.reporters.append(app.PDBReporter(filename, 1, atomSubset=atomSubset))
+
+            simulation.step(1)
+
+            # clear reporters to ensure PDBReporter calls writeFooter and file.close
+            simulation.reporters.clear()
+
+            # for PyPy the above line is not enough, we need to force garbage collection to ensure the 
+            # PDBReporter.__del__ method has been called before we open the file for reading 
+            gc.collect()
+
+            validpdb = pdb
+            testpdb = app.PDBFile(filename)
+
+            validBonds = list(validpdb.topology.bonds())
+            testBonds = list(testpdb.topology.bonds())
+            
+            self.assertEqual(len(validBonds), len(testBonds))
+
+            for validBond, testBond in zip(validBonds, testBonds):
+                self.assertEqual(str(validBond), str(testBond))
+
+
+    def assertVecAlmostEqual(self, p1, p2, tol=1e-7):
+        unit = p1.unit
+        p1 = p1.value_in_unit(unit)
+        p2 = p2.value_in_unit(unit)
+        scale = max(1.0, norm(p1),)
+        for i in range(3):
+            diff = abs(p1[i]-p2[i])/scale
+            self.assertTrue(diff < tol)
+
+
+
+
+class TestPDBxReporter(unittest.TestCase):
+    def setUp(self):
+        self.pdb = app.PDBFile('systems/alanine-dipeptide-explicit.pdb')
+        self.forcefield = app.ForceField('amber99sbildn.xml','tip3p.xml')
+        self.system = self.forcefield.createSystem(self.pdb.topology, nonbondedMethod=app.CutoffNonPeriodic, constraints=app.HBonds)
+
+    def testSubset(self):
+        """Test writing out a subset of atoms"""
+    
+        with tempfile.TemporaryDirectory() as tempdir:
+            filename = os.path.join(tempdir, 'temptraj.pdbx')
+            simulation = app.Simulation(self.pdb.topology, self.system, mm.LangevinMiddleIntegrator(300*unit.kelvin, 1.0/unit.picosecond, 0.002*unit.picoseconds))
+            simulation.context.setPositions(self.pdb.positions)
+
+            # just the alanine-dipeptide atoms
+            subset = list(range(0,22))
+
+            simulation.reporters.append(app.PDBxReporter(filename, 1, atomSubset=subset))
+            simulation.step(1)
+
+            # clear reporters to ensure PDBxReporter calls file.close
+            simulation.reporters.clear()
+
+            # check if the output out trajectory contains the expected number of atoms
+            checkpdb = app.PDBxFile(filename)
+            self.assertEqual(len(checkpdb.positions), len(subset))
+
+            # check the positions are correct
+            validPositions  = [simulation.context.getState(getPositions=True).getPositions()[i] for i in subset]
+            # round to 5 decimal places before comparing
+            for i in range(len(validPositions)):
+                validPositions[i] = [round(validPositions[i][j]._value, 5) for j in (0, 1, 2)]*unit.nanometer
```

Comment:
Does this actually make a difference?  `assertVecAlmostEqual()` ought to work just as well without rounding first.

Reply:
If I don't round the position to 4 decimal places for comparing to PDB input and 5 decimal places for PDBx then the assertion fails unless I start reducing the tolerance in assertVecAlmostEqual() to 1e-4 for PDB and 1e-5 for PDBx.
e.g. (1.00001,1.00001,1.00001)nm will be written to PDB file as (10.000, 10.000, 10.000) angstroms, which will fail in assertVecAlmostEqual(), the diff is 5.77e-5. 

Given that we have set the precision we are writing the PDB and PDBx files with, `%8.3f` and %10.4f` Angstrom, I think it is clearer to just round the more precise values from getState() before comparing them

Reply:
I see, makes sense.